src/packages/cli/src/cli.ts

```diff
@@ -157,7 +158,7 @@ if (argv.action === "start") {
     // instance), so we need to notify that we are ready.
     const isDetachedInstance = process.send !== undefined;
     if (isDetachedInstance) {
-      notifyDetachedInstanceReady();
+      notifyDetachedInstanceReady(port);
```

Comment:
< controversy >

Reply:
I don't like this. Sending an object has _minimal_ additional overhead in terms of complexity and performance, and makes this statement much more obvious:

```
child.on("message", port => {
  resolve(port as number);
});
```

```
child.on("message", message => {
  if (message.message === READ_MESSAGE) {
    resolve(message.port as number);
  }
});
```

Granted, this area is not _super_ complicated, and the surrounding code is quite explanatory, but :shrug:./

Keen to hear other people's opinions...

Reply:
If we can't trust that `message` is what we expect it to be (since we have to check against `READY_MESSAGE`), then do we also need to check the entire payload?

```
try {
  if (message && typeof message === "object" && message.message === READ_MESSAGE && typeof message.port === "number" && isValidPort(message.port)) {
    resolve(message.port);
  }
} catch {
  // ignore unexpected messages
}
function isValidPort(port: number): boolean {
   // ...
}
```

Reply:
After much discussions, I'm happy with the solution as is, with the updated comment in detach.ts